src/services/users.service.ts

```diff
@@ -2,32 +2,16 @@ import { Service } from 'typedi';
 import { compareSync, genSaltSync, hashSync } from 'bcrypt';
 import { getRepository } from 'typeorm';
 import { User } from '@entities/user.entity';
-import { JWTService } from '@services/jwt.service';
-import { AuthInterface, UserInterface } from '@interfaces';
 
 @Service()
 export class UsersService {
-  constructor(private readonly jwtService: JWTService) {}
 
   private readonly userRepository = getRepository<User>(User);
 
-  comparePassword(input: AuthInterface.IComparePasswordInput): boolean {
-    const { password, userPassword } = input;
+  comparePassword(password: string, userPassword: string): boolean {
     return compareSync(password, userPassword);
```

Comment:
Please do not use sync functions, as Node a is a single thread language that is going to block the main thread application if the requests using this functions scale. 

The better thing you can do here is to use asynchronous functions in order to execute this in a different queue and only reach the main thread when it's available to start working on new tasks.